attestation-service/attestation-service/src/rvps/mod.rs

```diff
@@ -27,59 +29,64 @@ pub mod grpc;
-    ///
-    /// Only used when feature `rvps-builtin` is enabled
-    #[serde(default = "String::default")]
-    pub store_type: String,
+    /// Address of remote RVPS. If this field is given, a remote RVPS will be connected to.
+    /// If this field is not given, a built-in RVPS will be used.
+    pub remote_addr: Option<String>,
 
-    /// The address of the remote RVPS server, e.g.
-    ///
-    /// - `http://127.0.0.1:50002`
-    ///
-    /// Only used when feature `rvps-rpc` is enabled
-    #[serde(default = "String::default")]
-    pub remote_addr: String,
+    /// This field will be used only if `remote_addr` is not given.
+    pub store_type: Option<String>,
+
+    /// This field will be used only if `remote_addr` is not given.
+    pub store_config: Option<Value>,
+}
+
+impl From<RvpsConfig> for RvpsCrateConfig {
+    fn from(val: RvpsConfig) -> RvpsCrateConfig {
+        let store_type = val.store_type.unwrap_or_else(|| {
+            warn!("No RVPS' `store_type` specified in AS config. Use `{DEFAULT_STORAGE_TYPE}` as default.");
+            String::from(DEFAULT_STORAGE_TYPE)
+        });
+        RvpsCrateConfig {
+            store_type,
+            store_config: val.store_config,
+        }
+    }
 }
 
 impl Default for RvpsConfig {
     fn default() -> Self {
         Self {
-            store_type: "LocalFs".into(),
-            remote_addr: Default::default(),
+            remote_addr: None,
+            store_type: Some(String::from(DEFAULT_STORAGE_TYPE)),
+            store_config: None,
         }
     }
 }
 
-impl RvpsConfig {
-    /// If remote addr is specified and the feature `rvps-grpc` is enabled when
-    /// built, will try to connect the remote rvps. Or, will use a built-in rvps.
-    pub async fn to_rvps(&self) -> Result<Box<dyn RvpsApi + Send + Sync>> {
-        cfg_if::cfg_if! {
-            if #[cfg(feature = "rvps-grpc")] {
-                if !self.remote_addr.is_empty() {
-                    info!("connect to remote RVPS: {}", self.remote_addr);
-                    Ok(Box::new(grpc::Agent::new(&self.remote_addr).await?) as Box<dyn RvpsApi + Send + Sync>)
-                } else {
-                    cfg_if::cfg_if! {
-                        if #[cfg(feature = "rvps-builtin")] {
-                            warn!("No RVPS address provided and will launch a built-in rvps");
-                            Ok(Box::new(builtin::Rvps::new(&self.store_type)?) as Box<dyn RvpsApi + Send + Sync>)
-                        } else {
-                            Err(anyhow!("either feature `rvps-grpc` or `rvps-builtin` should be enabled."))
-                        }
+pub async fn initialize_rvps_client(config: &RvpsConfig) -> Result<Box<dyn RvpsApi + Send + Sync>> {
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "rvps-grpc")] {
+            if config.remote_addr.as_ref().is_some_and(|addr| !addr.is_empty()) {
+                let remote_addr = config.remote_addr.as_ref().expect("must be Some");
```

Comment:
it's probably better to give a more descriptive panic message ("remote addr should be set" or something)

Reply:
Yes. I am refactoring this and try to avoid `unwrap`s